pkgs/test_core/lib/src/bootstrap/vm.dart

```diff
@@ -24,3 +27,30 @@ void internalBootstrapVmTest(Function Function() getMain, SendPort sendPort) {
     platformChannel.sink.add('done');
   });
 }
+
+/// Bootstraps a native executable test to communicate with the test runner over
+/// a socket.
+void internalBootstrapNativeTest(
+    Function Function() getMain, List<String> args) async {
+  if (args.length != 2) {
+    throw StateError(
+        'Expected exactly two args, a host and a port, but got $args');
+  }
+  var socket = await Socket.connect(args[0], int.parse(args[1]));
+  var platformChannel = MultiChannel<Object?>(StreamChannel(socket, socket)
+      .cast<List<int>>()
```

Comment:
Hmm, is this happening because we aren't getting upwards inference from the `socket` argument?

What happens if you assign it to a variable so inference has no context?

```suggestion
  var socketChannel = StreamChannel(socket, socket);
  var platformChannel = MultiChannel<Object?>(socketChannel
```

Or maybe fill in the generic explicitly instead of use a `.cast`?

```suggestion
  var platformChannel = MultiChannel<Object?>(StreamChannel<List<int>>(socket, socket)
```

Reply:
The issue is the underlying stream/sink are still typed with the `Uint8List` generic type. The only way I have figured out how to fix it so far is with the cast as I have it.

Reply:
(there is no static error in any of these cases, but it is a runtime error without the cast on the stream channel itself)

Reply:
The specific error I get is `type 'Utf8Decoder' is not a subtype of type 'StreamTransformer<Uint8List, String>' of 'streamTransformer'`

Reply:
I wonder if you can cast only the stream or only the sink argument?

It's not  a big deal either way - if the `cast` works I'm fine with keeping it.

Reply:
Keeping `cast`